Src/WitsmlExplorer.Frontend/components/Modals/TubularComponentPropertiesModal.tsx

```diff
@@ -50,89 +50,72 @@ const TubularComponentPropertiesModal = (props: TubularComponentPropertiesModalI
-                    </MenuItem>
-                  ))}
-                </Select>
-                <TextField
-                  id={"id"}
-                  label={"id"}
-                  type="number"
-                  fullWidth
-                  InputProps={{
-                    startAdornment: <InputAdornment position="start">{editableTubularComponent.id ? editableTubularComponent.id.uom : ""}</InputAdornment>
-                  }}
-                  disabled={!editableTubularComponent.id}
-                  value={editableTubularComponent.id?.value}
-                  onChange={(e) =>
-                    setEditableTubularComponent({
-                      ...editableTubularComponent,
-                      id: { value: parseFloat(e.target.value), uom: editableTubularComponent.id.uom }
-                    })
-                  }
-                />
-                <TextField
-                  id={"od"}
-                  label={"od"}
-                  type="number"
-                  fullWidth
-                  InputProps={{
-                    startAdornment: <InputAdornment position="start">{editableTubularComponent.od ? editableTubularComponent.od.uom : ""}</InputAdornment>
-                  }}
-                  disabled={!editableTubularComponent.od}
-                  value={editableTubularComponent.od?.value}
-                  onChange={(e) =>
-                    setEditableTubularComponent({
-                      ...editableTubularComponent,
-                      od: { value: parseFloat(e.target.value), uom: editableTubularComponent.od.uom }
-                    })
-                  }
-                />
-                <TextField
-                  id={"len"}
-                  label={"len"}
-                  type="number"
-                  fullWidth
-                  InputProps={{
-                    startAdornment: <InputAdornment position="start">{editableTubularComponent.len ? editableTubularComponent.len.uom : ""}</InputAdornment>
-                  }}
-                  disabled={!editableTubularComponent.len}
-                  value={editableTubularComponent.len?.value}
-                  onChange={(e) =>
-                    setEditableTubularComponent({
-                      ...editableTubularComponent,
-                      len: { value: parseFloat(e.target.value), uom: editableTubularComponent.len.uom }
-                    })
-                  }
-                />
-              </FormControl>
+              <Autocomplete
+                label="typeTubularComp"
+                initialSelectedOptions={[editableTubularComponent.typeTubularComponent]}
+                options={tubularComponentTypes}
+                onOptionsChange={({ selectedItems }) => {
+                  setEditableTubularComponent({ ...editableTubularComponent, typeTubularComponent: selectedItems[0] });
+                }}
+                hideClearButton={!!editableTubularComponent.typeTubularComponent}
```

Comment:
Given that typeTubularComponent is a required element in WITSML, it should be sufficient to set hideClearButton to true